ledger/blockqueue.go

```diff
@@ -306,11 +326,34 @@ func (bq *blockQueue) getBlockCert(r basics.Round) (blk bookkeeping.Block, cert
 		return
 	}
 
+	start := time.Now()
+	ledger_getblockcert_count.Inc(nil)
 	err = bq.l.blockDBs.rdb.Atomic(func(ctx context.Context, tx *sql.Tx) error {
 		var err0 error
 		blk, cert, err0 = blockGetCert(tx, r)
 		return err0
 	})
+	counterMicros(ledger_getblockcert_micros, start)
 	err = updateErrNoEntry(err, lastCommitted, latest)
 	return
 }
+
+// TODO: promote this to a utility function in util/metrics ?
+func counterMicros(counter *metrics.Counter, start time.Time) {
+	counter.AddUint64(uint64(time.Now().Sub(start).Microseconds()), nil)
+}
```

Comment:
Yes, add a method to counter that would do so. maybe `AddDurationSinceMicroseconds(..)` ?